htmx-spring-boot/src/main/java/io/github/wimdeblauwe/htmx/spring/boot/mvc/HtmxHandlerInterceptor.java

```diff
@@ -94,10 +96,19 @@ private void setHxLocation(HttpServletResponse response, Method method) {
         }
     }
 
-    private void setHxPushUrl(HttpServletResponse response, Method method) {
+    private void setHxPushUrl(HttpServletRequest request, HttpServletResponse response, Method method) {
         HxPushUrl methodAnnotation = AnnotatedElementUtils.findMergedAnnotation(method, HxPushUrl.class);
         if (methodAnnotation != null) {
-            setHeader(response, HX_PUSH_URL, methodAnnotation.value());
+            if(!Objects.equals(methodAnnotation.value(), HtmxValue.TRUE)){
+                setHeader(response, HX_PUSH_URL, methodAnnotation.value());
+            }
+            String path = request.getRequestURI();
+            String queryString = request.getQueryString();
+
+            if (queryString != null && !queryString.isEmpty()) {
+                path += "?" + queryString;
+            }
+            setHeader(response, HX_PUSH_URL, path);
```

Comment:
You should use an if/else and also do the same in `setHxReplaceUrl`.

```java
 private void setHxPushUrl(HttpServletRequest request, HttpServletResponse response, Method method) {
    HxPushUrl methodAnnotation = AnnotatedElementUtils.findMergedAnnotation(method, HxPushUrl.class);
    if (methodAnnotation != null) {
        if (HtmxValue.TRUE.equals(methodAnnotation.value())) {
            setHeader(response, HX_PUSH_URL, getRequestUrl(request));
        } else {
            setHeader(response, HX_PUSH_URL, methodAnnotation.value());
        }
    }
}

private void setHxReplaceUrl(HttpServletRequest request, HttpServletResponse response, Method method) {
    HxReplaceUrl methodAnnotation = AnnotatedElementUtils.findMergedAnnotation(method, HxReplaceUrl.class);
    if (methodAnnotation != null) {
        if (HtmxValue.TRUE.equals(methodAnnotation.value())) {
            setHeader(response, HX_REPLACE_URL, getRequestUrl(request));
        } else {
            setHeader(response, HX_REPLACE_URL, methodAnnotation.value());
        }
    }
}

private String getRequestUrl(HttpServletRequest request) {
    return ServletUriComponentsBuilder.fromRequest(request).toUriString();
}
```

Reply:
`ServletUriComponentsBuilder.fromRequest(request).toUriString();` creates an absolute URL:
<img width="1237" alt="image" src="https://github.com/user-attachments/assets/bd7eefee-f2f4-499a-ab24-fe5db9d34a3d">

I think it would be better to create a relative URL as we don't know if the server runs behind some kind of proxy?

Reply:
I agree, if a relative URL works, then I think this is the best.

Reply:
I also agree 👍